framework/src/actions/ElementIDOutputAction.C

```diff
@@ -55,7 +56,7 @@ ElementIDOutputAction::act()
         auto var_params = _factory.getValidParams("MooseVariableConstMonomial");
         auto kernel_params = _factory.getValidParams("ExtraElementIDAux");
         kernel_params.set<ExecFlagEnum>("execute_on") = EXEC_INITIAL;
-        for (unsigned int i = 0; i < _mesh->getMesh().n_elem_integers(); ++i)
+        for (unsigned int i = 0; i < _problem->assembly(0, 0).numExtraElemIntegers(); ++i)
```

Comment:
Is this change necessary? It seems like this PR could be logically split in two. The first would be the added assertion checking ensuring that someone doesn't get an invalid ID from `Assembly`. That's definitely a needed change.

I'm less sure about this one. The name of this class is not `ElementIDsHeldByAssemblyOutputAction`. Do you need the ids output here to be consistent with the ids help by `Assembly`? Now this output action is inconsistent with the extra elem ids held by the *Mesh*

Reply:
never mind. I see the issue since we go through assembly for the aux kernel

Reply:
ExtraElementIDAux this action adds requires Assembly.